modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -150,6 +150,7 @@ package object subschema {
   def isSubType(s1: Schema, s2: Schema): Compatibility = (s1, s2) match {
     case (_, `any`)                                                        => Compatible
     case (`none`, _)                                                       => Compatible
+    case (s1, s2) if !s1.anyOf.isDefined && s2.anyOf.isDefined             => anyOfSubType(s1, s2)
```

Comment:
It seems that we have `!s1.anyOf.isDefined && s2.anyOf.isDefined => anyOfSubType(s1, s2)` and then `s1.anyOf.isDefined && s2.anyOf.isDefined => anyOfSubType(s1, s2)`, so both of those lines can be simplified into `s2.anyOf.isDefined => anyOfSubType(s1, s2)`

It is semantically different because of the order, but looking at it seems that it should be fine to do so. 
```suggestion
    case (s1, s2) if s2.anyOf.isDefined             => anyOfSubType(s1, s2)
```

and remove line 1

Reply:
nice, your logic simplification eyes are 20/20